ffmpeg link error

**Describe the bug**
ffmpeg can not link to executable file

**Environment**
- OS: [e.g. Windows/Linux etc...] Linux x64 
- Compiler: gcc8.3

**To Reproduce**
Steps to reproduce the behavior:
cmake file
```
cmake_minimum_required(VERSION 3.15)

project(usb_cam)

set(CMAKE_CXX_STANDARD 11)

if (CMAKE_SYSTEM_NAME MATCHES "Linux")
    message(STATUS "current platform: Linux ")
    set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O2" )
    find_package(FFMPEG REQUIRED)
    include_directories(${FFMPEG_INCLUDE_DIRS})
    link_directories(${FFMPEG_LIBRARY_DIRS})
elseif (CMAKE_SYSTEM_NAME MATCHES "Windows")
    message(STATUS "current platform: Windows")
else ()
    message(STATUS "other platform: ${CMAKE_SYSTEM_NAME}")
endif (CMAKE_SYSTEM_NAME MATCHES "Linux")

include_directories(src)

file(GLOB_RECURSE SRC_FILES src/*.cpp *.h)

add_executable(usb_cam ${SRC_FILES})
target_link_libraries(usb_cam ${FFMPEG_LIBRARYS})

cpp file
#include <iostream>
using namespace std;
extern "C"
{
#include <math.h>
#include <libavutil/opt.h>
#include <libavcodec/avcodec.h>
#include <libavutil/channel_layout.h>
#include <libavutil/common.h>
#include <libavutil/imgutils.h>
#include <libavutil/mathematics.h>
#include <libavutil/samplefmt.h>
#include <libavformat/avformat.h>
#include <libavdevice/avdevice.h>
#include <libavutil/dict.h>
};
int main(int argc, char **argv)
{
    int ret;
    AVFormatContext *fmtCtx = NULL;
    AVPacket pkt1, *pcaket = &pkt1;
    avcodec_register_all();
    avdevice_register_all();
    AVInputFormat *inputFmt = av_find_input_format("video4linux2");
    if (NULL != inputFmt)
    {
        std::cout << "input device name:" << inputFmt->name << std::endl;
    }
    else
    {
        std::cout << "Null point!" << std::endl;
    }
    ret = avformat_open_input(&fmtCtx, "/dev/video0", inputFmt, NULL);
    if (0 == ret)
    {
        std::cout << "Open input device seccess!" << std::endl;
    }
    av_read_frame(fmtCtx, pcaket);
    std::cout << "packet size:" << (pcaket->size) << std::endl;
    FILE *fp = NULL;
    fp = fopen("out.yuv", "wb");
    if (NULL != fp)
    {
        fwrite(pcaket->data, 1, pcaket->size, fp);
    }
    fclose(fp);
    av_free_packet(pcaket);
    avformat_close_input(&fmtCtx);
    return 0;
}
```
**Expected behavior**
A clear and concise description of what you expected to happen.

**Failure logs** 
```
main.cpp:(.text.startup+0x12): undefined reference to `avcodec_register_all'
/usr/bin/ld: main.cpp:(.text.startup+0x17): undefined reference to `avdevice_register_all'
/usr/bin/ld: main.cpp:(.text.startup+0x23): undefined reference to `av_find_input_format'
/usr/bin/ld: main.cpp:(.text.startup+0x77): undefined reference to `avformat_open_input'
/usr/bin/ld: main.cpp:(.text.startup+0x91): undefined reference to `av_read_frame'
/usr/bin/ld: main.cpp:(.text.startup+0x103): undefined reference to `av_free_packet'
/usr/bin/ld: main.cpp:(.text.startup+0x10b): undefined reference to `avformat_close_input'
collect2: error: ld returned 1 exit status
```

**Additional context**
Add any other context about the problem here.

--- a/ports/ffmpeg/CONTROL
+++ b/ports/ffmpeg/CONTROL
@@ -1,5 +1,5 @@
 Source: ffmpeg
-Version: 4.2-3
+Version: 4.2-4
 Build-Depends: zlib
 Homepage: https://ffmpeg.org
 Description: a library to decode, encode, transcode, mux, demux, stream, filter and play pretty much anything that humans and machines have created.

--- a/ports/ffmpeg/FindFFMPEG.cmake.in
+++ b/ports/ffmpeg/FindFFMPEG.cmake.in
@@ -156,10 +156,10 @@ if (FFMPEG_libavcodec_FOUND AND FFMPEG_libavdevice_FOUND AND FFMPEG_libavfilter_
     ${FFMPEG_libavdevice_LIBRARY}
     ${FFMPEG_libavfilter_LIBRARY}
     ${FFMPEG_libavformat_LIBRARY}
-    ${FFMPEG_libavresample_LIBRARY}
-    ${FFMPEG_libswresample_LIBRARY}
     ${FFMPEG_libswscale_LIBRARY}
     ${FFMPEG_libavcodec_LIBRARY}
+    ${FFMPEG_libswresample_LIBRARY}
+    ${FFMPEG_libavresample_LIBRARY}
     ${FFMPEG_libavutil_LIBRARY} 
     ${FFMPEG_libzlib_LIBRARY}
     ${FFMPEG_PLATFORM_DEPENDENT_LIBS}

--- a/ports/ffmpeg/FindFFMPEG.cmake.in
+++ b/ports/ffmpeg/FindFFMPEG.cmake.in
@@ -153,14 +153,14 @@ if (FFMPEG_libavcodec_FOUND AND FFMPEG_libavdevice_FOUND AND FFMPEG_libavfilter_
   set(FFMPEG_libswscale_VERSION "${FFMPEG_VERSION}" CACHE STRING "")
 
   list(APPEND FFMPEG_LIBRARIES
-    ${FFMPEG_libavformat_LIBRARY}
     ${FFMPEG_libavdevice_LIBRARY}
-    ${FFMPEG_libavcodec_LIBRARY}
-    ${FFMPEG_libavutil_LIBRARY}
-    ${FFMPEG_libswscale_LIBRARY}
     ${FFMPEG_libavfilter_LIBRARY}
-    ${FFMPEG_libswresample_LIBRARY}
+    ${FFMPEG_libavformat_LIBRARY}
     ${FFMPEG_libavresample_LIBRARY}
+    ${FFMPEG_libswresample_LIBRARY}
+    ${FFMPEG_libswscale_LIBRARY}
+    ${FFMPEG_libavcodec_LIBRARY}
+    ${FFMPEG_libavutil_LIBRARY} 
     ${FFMPEG_libzlib_LIBRARY}
     ${FFMPEG_PLATFORM_DEPENDENT_LIBS}
   )

